tests/src/HistoryMiddleware.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\Tests\EuropaSearchClient;
+
+use GuzzleHttp\Middleware;
+
+/**
+ * An HTTP client middleware that wraps the Guzzle history middleware.
+ *
+ * Note: this service has state! It should be only used for test purposes.
+ * It will keep track of the history of requests and responses across all the
+ * HTTP client instances.
+ */
+class HistoryMiddleware
```

Comment:
This was needed in Drupal, to register as a middleware in the Drupal client stack. But here there's no need for this at all. Also, it's clearly not a service.

Reply:
done